### PR TITLE
stage add: unsuppress -c/--checkpoints help text

### PR DESCRIPTION
**Pull request by pmrowla**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Related to https://github.com/iterative/dvc.org/issues/2253

docs PR: https://github.com/iterative/dvc.org/pull/2290

### CODE REVIEW

File: dvc/command/stage.py [hunk 1]
```diff
@@ -255,7 +255,8 @@ def _add_common_args(parser):
         "--checkpoints",
         action="append",
         default=[],
-        help=argparse.SUPPRESS,
+        help="Declare output file or directory that will be be handled as an "
+        "intermediate checkpoint upon exp run.",
```
Comment (dberenbaum) [inline dvc/command/stage.py:259]: Any reason to leave out "Note that checkpoints are not supported by `dvc repro`."?

**Review by dberenbaum — APPROVED** [review]
One small question, but looks good!

File: dvc/command/stage.py (hunk 1 above)
Comment (jorgeorpinel) [reply dvc/command/stage.py:259]: Does `repro` error-out or just ignore them?

Comment (dberenbaum) [reply dvc/command/stage.py:259]: Errors out:

```
ERROR: failed to reproduce 'dvc.yaml': Checkpoint stages are not supported in 'dvc repro'. Checkpoint stages must be reproduced with 'dvc exp run' or 'dvc exp resume'.
```

**Comment by jorgeorpinel** [comment]
How does this work, something like

```console
$ dvc stage add -n mystage -d mydata -d myout1 -c myout2 ...
```

? I.e. `-c` replaces `-o`

File: dvc/command/stage.py (hunk 1 above)
Comment (jorgeorpinel) [reply dvc/command/stage.py:259]: OK my suggested text: ``"Declare checkpoint output (file or directory) for `dvc exp run`. Not compatible with `dvc repro`"``

**Comment by pmrowla** [comment]
> How does this work, something like
> 
> ```
> $ dvc stage add -n mystage -d mydata -o myout1 -c myout2 -M ...
> ```
> 
> ? I.e. `-c` replaces `-o`

Yes, it works the same way as the metrics flags or `--outs-persist` for generating a specific type of output

**Pull request merged**